### PR TITLE
Raise an explicit error for zips uploads with an unsupported compression method

### LINKED ISSUE #19814: NotImplementedError: That compression method is not supported
When a developer submits a zipfile with an unsupported compression method, we should handle the error and return an appropriate validation message.

This notably happens with zip files compressed with the `PPMd`, `xz` or `DEFLATE64` methods (but with a `.zip` filename), which are unsupported by Python.

Sentry Issue: [ADDONS-SERVER-PROD-8T](https://sentry.io/organizations/mozilla/issues/3688396712/?referrer=github_integration)

```
NotImplementedError: That compression method is not supported
(6 additional frame(s) were not displayed)
...
  File "zipfile.py", line 1698, in _extract_member
    with self.open(member, pwd=pwd) as source, \
  File "zipfile.py", line 1571, in open
    return ZipExtFile(zef_file, mode, zinfo, pwd, True)
  File "zipfile.py", line 800, in __init__
    self._decompressor = _get_decompressor(self._compress_type)
  File "zipfile.py", line 699, in _get_decompressor
    _check_compression(compress_type)
  File "zipfile.py", line 679, in _check_compression
    raise NotImplementedError("That compression method is not supported")

Unhandled error during validation: NotImplementedError('That compression method is not supported')
```

### PR DESCRIPTION
**Pull request by diox**
Fixes #19814

### CODE REVIEW

File: src/olympia/files/tests/test_utils.py [hunk 1]
```diff
@@ -1216,65 +1217,122 @@ def test_raises_error_for_absolute_path(self):
...
-            utils._validate_archive_member_name_and_size(None, 123)
+            utils.archive_member_validator(info)
+
+    def test_unsupported_compression_method(self):
+        info = self._fake_archive_member('foo', 123)
+        info.compress_type == zipfile.ZIP_DEFLATED
+        # ZIP_DEFLATED works.
+        utils.archive_member_validator(info)
+        # ZIP_STORED works.
+        info.compress_type = zipfile.ZIP_DEFLATED
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_utils.py:1238]: This is also setting `compress_type` to `zipfile.ZIP_DEFLATED`.

File: src/olympia/files/tests/test_utils.py [hunk 2]
```diff
@@ -1216,65 +1217,122 @@ def test_raises_error_for_absolute_path(self):
...
+        for compress_type in (
+            ZIP_DEFLATED64,
+            ZIP_IMPLODED,
+            ZIP_BZIP2,
+            ZIP_LZMA,
+            ZIP_ZSTANDARD,
+            ZIP_XZ,
+            ZIP_PPMD,
+        ):
+            info.compress_type = zipfile.ZIP_LZMA
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_utils.py:1258]: You're always setting `compress_type` to `zipfile.ZIP_LZMA`, as opposed to using `compress_type` from the loop.

**Review by bobsilverberg — APPROVED** [review]
Looks good, but I think there are a couple of problems with the tests.

r+wc

File: src/olympia/files/tests/test_utils.py (hunk 2 above)
Comment (diox) [reply src/olympia/files/tests/test_utils.py:1258]: doh, I added the loop late, didn't re-read it. I'll fix it.

File: src/olympia/files/tests/test_utils.py (hunk 1 above)
Comment (diox) [reply src/olympia/files/tests/test_utils.py:1238]: :man_facepalming:

File: src/olympia/files/tests/test_utils.py (hunk 2 above)
Comment (diox) [reply src/olympia/files/tests/test_utils.py]: ead5ccc56acda3b3a22a6b845d9e23a9c1fe91c2

File: src/olympia/files/tests/test_utils.py (hunk 1 above)
Comment (diox) [reply src/olympia/files/tests/test_utils.py]: ead5ccc56acda3b3a22a6b845d9e23a9c1fe91c2

**Pull request merged**